libraries/chain/backing_store/tests/generate_keys.cpp

```diff
@@ -0,0 +1,117 @@
+#include<iostream>
+#include<fstream>
+
+namespace gen_keys {
+
+struct cmd_args {
+   uint32_t num_keys = 0;
+   std::string first_key;
+   std::string key_file;
+};
+
+// Generates "num_keys" keys starting with "first_key" alphabetically
+// and saves them into "key_file".
+// One may use this program to generate multiple files of keys,
+// concatenate them, and use Linux shuf command to randomize the
+// resulting file.
+
+void print_usage(char* program_name) {
+   std::cout << program_name << " - Generates \"num_keys\" keys starting with \"first_key\" alphabetically and saves them into \"key_file\"\n\n"
+      << "   -f=FILENAME, file storing the resultingkeys\n"
+      << "   -n=NUMBER, number of keys to be generated\n"
+      << "   -k=KEY, the first key to be generated\n"
+      << "   -h, print this help\n"
+      << std::endl;
+}
+
+// As this program is only for internal use,
+// we don't use sophisticated library to parse command line
+void parse_cmd_line(int argc, char* argv[], cmd_args& args) {
+   if (argc != 7) {
+      std::cout << "Require 3 arguments" << std::endl;
```

Comment:
We are checking for 7 arguments.

Reply:
What I meant we needed key file name, first key, and number of keys. Will reword.